app/services/unified_price_service.py

```
@@ -262,21 +262,180 @@ async def _fetch_batch_from_source(self, symbols: List[str], source: PriceSource
+                self.logger.info(f"KuCoin direct fallback successful", symbol=symbol, price=price)
+                return float(price)
+                
+        except Exception as e:
+            self.logger.warning("Exchange direct fallback failed", symbol=symbol, error=str(e))
+        
+        # 4. ENTERPRISE: Additional free APIs (no limits)
+        try:
+            # CoinGecko simple API (no key required)
+            import aiohttp
+            async with aiohttp.ClientSession() as session:
+                url = f"https://api.coingecko.com/api/v3/simple/price"
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)